server/route.go

```diff
@@ -659,12 +659,13 @@ func (s *Server) routeAcceptLoop(ch chan struct{}) {
 	}
 
 	hp := net.JoinHostPort(opts.Cluster.Host, strconv.Itoa(port))
-	s.Noticef("Listening for route connections on %s", hp)
 	l, e := net.Listen("tcp", hp)
 	if e != nil {
-		s.Fatalf("Error listening on router port: %d - %v", opts.Cluster.Port, e)
+		s.Fatalf("Error listening on router port: %d - %v", l.Addr().(*net.TCPAddr).Port, e)
 		return
 	}
+	s.Noticef("Listening for route connections on %s",
```

Comment:
Since we need that info again below, you could do this:
```
realPort := port
if port == 0 {
  realPort = l.Addr().(*net.TCPAddr).Port
}
s.Noticef(... strconv.Itoa(realPort)))
```
and then down below
```
if port == 0 {
  // Write resolved port back to options.
  opts.Cluster.Port = realPort
}
```